csharp/ql/integration-tests/all-platforms/cshtml/global.json

```diff
@@ -0,0 +1,5 @@
+{
+    "sdk": {
+        "version": "7.0.102"
```

Comment:
Should we add `"rollForward": "latestFeature"`, this would mean that any newer 7.0 variant would be accepted too.

Reply:
I would prefer that we concretely pin the version number and not even accept another version with the same feature band to avoid switching to a newer version, when it is installed on the runner. At least for .NET 7 we have around 5 different versions installed and I am uncertain that it is the same across all platforms - so I will prefer that it is completely under our control when a newer version is used even with the same feature band.

Reply:
I'm okay with this decision as I don't fully understand the problem that we're solving. Is it that previously the latest `7.0.x` version of the SDK was used, while now it's the latest `8.0.x` version? (Unless we somehow force the usage of a `7.0.x` variant, which this PR is doing.)

I think version `7.0.102` is the version that we're installing to the runners. The [default installed variants](https://github.com/search?q=repo%3Aactions%2Frunner-images+%22.NET+Core+SDK%22+Readme.md&type=code) seem to depend on the OS image. So if we were to upgrade our jobs to use `7.0.404`, then we would have to update all the tests too, wouldn't we?

Reply:
Answering for @michaelnebel, but I think the problem we are solving with this is the general case of what you are describing. When the GHA images get updated to a new .NET version, but we do not yet support that .NET version, then this may cause widespread test failures throughout not just the `main` branch, but also `rc` branches etc. (as we have seen with the recent .NET 8 release). Pinning the version guarantees that there's a match between the version that the tests are designed for in a given branch and the version of .NET that is used there.

I think your point about not necessarily having the right version installed, depending on the runner image, is a valid one. Perhaps we should add `setup-dotnet` steps with the same version, just to be sure.

For Go, we do allow the patch version to differ (i.e. we have it pinned to `1.21.x` right now for any `x`). I think it is probably OK to similarly pin .NET to `7.0.x` since we wouldn't expect patch-level releases to introduce any breaking changes.

Reply:
@tamasvajk : Yes, our CI has used the most recent installed SDK (which meant 7.0.404) until 8.0.100 was installed - then this was used by default.
The reason why I picked 7.0.102 is because that is the one we are installing on the runners as a part of setup-dotnet - so to make it consistent I picked that (I could also have forced the installation of the newer version). The .NET version has just changed gradually over time during the last year as newer versions of .NET became available on the runners.
Also, I don't think we should be to concerned whether we pick 7.0.102 or 7.0.404 for this change as the version number will change again as part of the .NET 8 update PR(s) and be pinned to a .NET 8 versions.